src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Form/Entity/OidcngResourceServerOptionsFactory.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Form\Entity;
+
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Exception\InvalidEnvironmentException;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Client\QueryClient;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Dto\ManageEntity;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Exception\QueryServiceProviderException as QueryServiceProviderExceptionAlias;
+
+class OidcngResourceServerOptionsFactory
+{
+    /**
+     * @var QueryClient
+     */
+    private $testEntityRepository;
+
+    /**
+     * @var QueryClient
+     */
+    private $productionRepository;
+
+    private $testPublicationState;
+
+    private $productionPublicationState;
+
+    /**
+     * @param QueryClient $testEntityRepository
+     * @param QueryClient $productionRepository
+     * @param string $testPublicationState
+     * @param string $productionPublicationState
+     */
+    public function __construct(
+        QueryClient $testEntityRepository,
+        QueryClient $productionRepository,
+        $testPublicationState,
+        $productionPublicationState
+    ) {
+        $this->testEntityRepository = $testEntityRepository;
+        $this->productionRepository = $productionRepository;
+        $this->testPublicationState = $testPublicationState;
+        $this->productionPublicationState = $productionPublicationState;
+    }
+
+
+    /**
+     * @param string $teamName
+     * @param string $environment
+     * @return array
```

Comment:
Why not return a `ResourceServerCollection`?

Reply:
I think it would be better if that isn't the responsibility of the factory. The consumer would be able to use the `ResourceServerCollection::getResourceServers` method to get an array while the factory will be reusable by other consumer later on. Else you would have to create a factory dependent on used consumers. I don't mind if you leave it for now, but you could consider returning a VO next time.

Reply:
I don't see this being reused. And if it would be, then it would most certainly be in the context of ChoiceType options generation (as this is the single responsibility of this factory), and that is what it would be able to do just fine. 

Also, introducing an additional complexity for the sake of possible reusability down the road is not something I consider valuable here.

Maybe you are aiming at something else here?